source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -356,6 +356,11 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
     NOT_REACHED_GCOVR_EXCL_LINE;
   }
 
+  envoy::config::filter::network::http_connection_manager::v2::HttpFilter message;
+  message.set_name("envoy.on_demand");
+  bool terminal_filter = false;
+  processFilter(message, -1, "http", filter_factories_, terminal_filter);
```

Comment:
Why isn't this being explicitly configured? Some folks might not want to support VHDS.

Reply:
VHDS is only active if it is enabled in the route configuration. 

If I made the filter explicitly configurable, a user would need to:
 - configure vhds in the route config,
 - add on_demand filter to the filter chain
  - and placed in the correct position (before router)

Isn't it less error-prone if it's there automatically?

Reply:
Generally we try and avoid having magic changes to configuration under the hood, since there is a tradeoff between being less error-prone and being more surprising. So, I'd go for principle of least surprise, and make this explicit but well documented.